pkg/streamer/reader.go

```diff
@@ -71,12 +74,15 @@ func NewBinlogReader(cfg *BinlogReaderConfig) *BinlogReader {
 	if cfg.Timezone != nil {
 		parser.SetTimestampStringLocation(cfg.Timezone)
 	}
+
+	newtctx := tcontext.NewContext(ctx, tctx.L().WithFields(zap.String("component", "binlog reader")))
+
 	return &BinlogReader{
 		cfg:       cfg,
 		parser:    parser,
 		indexPath: path.Join(cfg.RelayDir, utils.UUIDIndexFilename),
-		ctx:       ctx,
 		cancel:    cancel,
+		tctx:      newtctx,
```

Comment:
we do not use `ctx` created in L69 anymore, so `cancel` will not work, right?

Reply:
we put `ctx` created in L69 into `newtctx` at L78